app/components/curriculum-inventory-report-list.js

```diff
@@ -16,74 +17,69 @@ const ReportProxy = ObjectProxy.extend({
   isPublished: alias('isFinalized'),
 
   userCanDelete: computed('content', 'content.isFinalized', 'currentUser.model', async function() {
-    const permissionChecker = this.permissionChecker;
     const report = this.content;
-    if (report.get('isFinalized')) {
+    if (report.isFinalized) {
       return false;
     }
-    return permissionChecker.canDeleteCurriculumInventoryReport(report);
+    return this.permissionChecker.canDeleteCurriculumInventoryReport(report);
   })
 });
 
-export default Component.extend({
-  currentUser: service(),
-  intl: service(),
-  permissionChecker: service(),
-  tagName: "",
-  program: null,
-  sortBy: 'name',
-  edit() {},
-  remove() {},
-  setSortBy() {},
+export default class CurriculumInventoryReportListComponent extends Component {
+  @service currentUser;
+  @service intl;
+  @service permissionChecker;
+  @tracked proxiedReports;
 
-  /**
-   * @property proxiedReports
-   * @type {Ember.computed}
-   * @public
-   */
-  proxiedReports: computed('program.curriculumInventoryReports.[]', async function () {
-    const currentUser = this.currentUser;
-    const intl = this.intl;
-    const permissionChecker = this.permissionChecker;
-    const program = this.program;
-    const reports = await program.get('curriculumInventoryReports');
-    return reports.map(report => {
+  @action
+  load(element, [ reports ]) {
```

Comment:
Since this is sync and only setting the one property couldn't it just be a getter?  `get proxiedReports()`?

Reply:
handled. i rewired the component to take the reports as input directly.